### PR TITLE
Add Blue Score to seq2seq metrics

### LINKED ISSUE #1301: [Metrics] Bleu
## 🚀 Feature
Implement BLEU

### PR DESCRIPTION
**Pull request by fmcurti**
## What does this PR do?

This PR adds a new module inside Metrics: seq2seq
and adds a PL Metric wrapper for torchtext bleu_score
Fixes #1301

# Before submitting

- [Y] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [Y] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [Y] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [Y] Did you make sure to update the documentation with your changes?
- [Y] Did you write any new necessary tests? 
- [Y] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun? 
Absolutely!

### CODE REVIEW

File: docs/source/metrics.rst [hunk 1]
```diff
@@ -228,6 +228,12 @@ RMSLE
 .. autoclass:: pytorch_lightning.metrics.regression.RMSLE
     :noindex:
 
+Bleu
+^^^^^
```
Comment (Borda) [inline docs/source/metrics.rst:232]:
```suggestion
^^^^
```

**Comment by fmcurti** [comment]
> Great! mind add it also as functional?

Sure thing! I  added a commit with the functional version and replaced the Metrics call with the functional one

**Comment by williamFalcon** [comment]
seq2seq is too sepcific. can you just call it nlp?  

from pytorch_lightning.metrics.nlp import xxxxx

**Comment by ydcjeff** [comment]
I have already worked on this bleu metric tho. Mind checking this PR #2535?

**Comment by Borda** [comment]
> I have already worked on this bleu metric tho. Mind checking this PR #2535?

sorry, you are right, lets get done PR #2535 first...

**Comment by fmcurti** [comment]
> I have already worked on this bleu metric tho. Mind checking this PR #2535?

ohh I'm sorry, I thought I saw no linked PR's to the issue.

**Comment by williamFalcon** [comment]
yeah makes sense. Let’s finish bleu in the other PR? but it’s great we have the two PRs so we can make sure to use the best ideas from both

**Comment by williamFalcon** [comment]
Added in #2535

**Pull request closed without merging**